File: apps/launcher/src/types/library.ts
export interface VersionSummary {
  id: string;
  kind: string;
}

export interface VersionManifestSummary {
  latestRelease: string;
  versions: VersionSummary[];
}

export interface FabricLoaderVersion {
  version: string;
  stable: boolean;
}

export type NeoForgeLoaderVersion = string;

export interface ModEntry {
  fileName: string;
  displayName: string;
  enabled: boolean;
  size: number;
  modified: number;
}

export type AtlasChannel = "dev" | "beta" | "production";

export interface AtlasRemotePack {
  packId: string;
  packName: string;
  packSlug: string;
  accessLevel: AtlasChannel;
  channel: AtlasChannel;
  buildId?: string | null;
  buildVersion?: string | null;
  artifactKey?: string | null;
}

export interface AtlasPackSyncResult {
  packId: string;
  channel: AtlasChannel;
  buildId?: string | null;
  buildVersion?: string | null;
  minecraftVersion?: string | null;
  modloader?: string | null;
  modloaderVersion?: string | null;
  bundledFiles: number;
  hydratedAssets: number;
}

File: apps/launcher/src-tauri/src/models/library.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Serialize, Deserialize, Clone)]
#[serde(rename_all = "camelCase")]
pub struct VersionSummary {
    pub id: String,
    pub kind: String,
}

#[derive(Debug, Serialize, Deserialize, Clone)]
#[serde(rename_all = "camelCase")]
pub struct VersionManifestSummary {
    pub latest_release: String,
    pub versions: Vec<VersionSummary>,
}

#[derive(Debug, Serialize, Deserialize, Clone)]
#[serde(rename_all = "camelCase")]
pub struct FabricLoaderVersion {
    pub version: String,
    pub stable: bool,
}

#[derive(Debug, Serialize, Deserialize, Clone)]
#[serde(rename_all = "camelCase")]
pub struct ModEntry {
    pub file_name: String,
    pub display_name: String,
    pub enabled: bool,
    pub size: u64,
    pub modified: u64,
}

#[derive(Debug, Serialize, Deserialize, Clone)]
#[serde(rename_all = "camelCase")]
pub struct AtlasRemotePack {
    pub pack_id: String,
    pub pack_name: String,
    pub pack_slug: String,
    pub access_level: String,
    pub channel: String,
    #[serde(default)]
    pub build_id: Option<String>,
    #[serde(default)]
    pub build_version: Option<String>,
    #[serde(default)]
    pub artifact_key: Option<String>,
    #[serde(default)]
    pub minecraft_version: Option<String>,
    #[serde(default)]
    pub modloader: Option<String>,
    #[serde(default)]
    pub modloader_version: Option<String>,
}

#[derive(Debug, Serialize, Deserialize, Clone)]
#[serde(rename_all = "camelCase")]
pub struct AtlasPackSyncResult {
    pub pack_id: String,
    pub channel: String,
    #[serde(default)]
    pub build_id: Option<String>,
    #[serde(default)]
    pub build_version: Option<String>,
    #[serde(default)]
    pub minecraft_version: Option<String>,
    #[serde(default)]
    pub modloader: Option<String>,
    #[serde(default)]
    pub modloader_version: Option<String>,
    #[serde(default)]
    pub force_reinstall: bool,
    #[serde(default)]
    pub requires_full_reinstall: bool,
    pub bundled_files: u64,
    pub hydrated_assets: u64,
}
